### PR TITLE
LoadNexusProcessed use int instead of int64_t

### LINKED ISSUE #16923: In LoadNexusProcessed use int instead of int64_t
Currently `LoadNexusProcessed` uses int64_t for all of it's properties. However it checks the boundaries up to `Mantid::EMPTY_INT()` which returns the maximum value for an int meaning we cannot use the additional range provided.

We should go through and change these int64_t to int in the algorithm properties.

### PR DESCRIPTION
**Pull request by DavidFair**
Description of work.
Previously `LoadNexusProcessed` used int64_t for its spectra properties and throughout the file. However the upper bound was capped to the range of an int wasting all the additional space of the 64 bit int. 
Most of the int64_t has now been converted to either int or size_t correctly with one exception in a function where it is needed. 
**To test:**
All unit tests should pass. Loading processed nexus files should still work. 

Fixes #16923 .
Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
Why is there a file `Framework/DataHandling/src/LoadNexusProcessed.cpp.orig`? Isn't that what revision control systems do by default?

**Comment by DavidFair** [comment]
@peterfpeterson This was an unwanted backup from the merge tool as the settings weren't set correctly. Thanks for spotting it.

File: Framework/DataHandling/src/LoadNexusProcessed.cpp [hunk 1]
```diff
@@ -289,13 +289,15 @@ Workspace_sptr LoadNexusProcessed::doAccelleratedMultiPeriodLoading(
...
+      static_cast<int>(nHistograms) /
+      blockSize; // Truncated number of full blocks to read. Remainder removed
   const int readOptimumStop = (nFullBlocks * blockSize);
   const int readStop = m_spec_max - 1;
   const int finalBlockSize = readStop - readOptimumStop;
 
   int wsIndex = 0;
-  //This cast is safe as m_spec_min must be >= 1
+  // This cast is safe as m_spec_min must be >= 1
   int histIndex = m_spec_min - 1;
```
Comment (KarlPalmen) [inline Framework/DataHandling/src/LoadNexusProcessed.cpp]: which cast?

**Comment by KarlPalmen** [comment]
I found one comment that may be referring to a cast that has been removed. I suggest you check the comments and ensure they are up to date.

**Comment by DavidFair** [comment]
@KarlPalmen I've gone through and checked for any additional comments which may be out of date. Thanks for spotting that.

**Comment by KarlPalmen** [comment]
Works OK with a few processed next files I have. :shipit:

**Pull request merged**